### PR TITLE
feat: added check for server_hostname

### LINKED ISSUE #4: Make server hostname configurable via environment variable
Add a check for environment variable named SERVER_HOSTNAME that allows you to specify the hostname the server will use.

Should default to `localhost`.

### PR DESCRIPTION
**Pull request by RohitShelke10**

### CODE REVIEW

**Review by pitchinnate — COMMENTED** [review]
`hostname` should not have `http://` in it. If you try to run this you will see an error:

```
[GIN-debug] [ERROR] listen tcp: address http://localhost:8080: too many colons in address
```

**Comment by RohitShelke10** [comment]
My bad

**Comment by RohitShelke10** [comment]
@pitchinnate Fixed it

File: cmd/server/main.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func serve() {
 	r := routes.SetupRouter()
 	hostname, ok = os.LookupEnv("SERVER_HOSTNAME")
 	if !ok {
-		hostname="http://localhost"
+		hostname="localhost"
 	}
 	r.Run("%s:8080", hostname)
```
Comment (pitchinnate) [inline cmd/server/main.go]: `r.Run("%s:8080", hostname)`

This isn't valid in Gin, it only accepts a single param. You could use `fmt.Sprintf`

**Comment by RohitShelke10** [comment]
@pitchinnate It is working now

**Pull request merged**